src/cockpit/internal_endpoints.py

```diff
@@ -77,15 +77,18 @@ def get_machines(self) -> Dict[str, Dict[str, Variant]]:
         results: Dict[str, Dict[str, Variant]] = {}
 
         for filename in glob.glob(f'{self.path}/*.json'):
-            with open(filename) as fp:
-                try:
-                    contents = json.load(fp)
-                except json.JSONDecodeError:
-                    logger.warning('Invalid JSON in file %s.  Ignoring.', filename)
-                    continue
-                # merge
-                for hostname, attrs in contents.items():
-                    results[hostname] = {key: Variant(value) for key, value in attrs.items()}
+            try:
+                with open(filename) as fp:
+                    try:
+                        contents = json.load(fp)
+                    except json.JSONDecodeError:
+                        logger.warning('Invalid JSON in file %s.  Ignoring.', filename)
+                        continue
+                    # merge
+                    for hostname, attrs in contents.items():
+                        results[hostname] = {key: Variant(value) for key, value in attrs.items()}
+            except OSError:
+                logger.error('Unreadable machines.d file %s.  Ignoring.', filename)
```

Comment:
Sorry, missed that last time: Please log the actual error. It might not actually be unreadable. I.e. `"Failed to open %s: %s", filename, e`